pkg/services/ngalert/store/provisioning_store.go

```diff
@@ -0,0 +1,81 @@
+package store
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+type provenanceRecord struct {
+	Id         int `xorm:"pk autoincr 'id'"`
+	OrgId      int
+	RecordKey  string
+	RecordType string
+	Provenance models.Provenance
+}
+
+func (pr provenanceRecord) TableName() string {
+	return "provenance_type"
+}
+
+type ProvisioningStore interface {
+	GetProvenance(o models.ProvisionedObject) (models.Provenance, error)
+	// TODO: API to query all provenances for a specific type?
```

Comment:
Can we get this in? This will be super important to support listing.

Reply:
Yes. The next PR in this chain also keeps this TODO (and expands it to the Transactional version of the API). I'll add it in a single pass in a future PR.